internal/pkg/agent/application/upgrade/marker_access_common_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package upgrade
+
+import (
+	"math/rand"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteMarkerFileWithTruncation(t *testing.T) {
+	tmpDir := t.TempDir()
+	testMarkerFile := filepath.Join(tmpDir, markerFilename)
+
+	// Write a long marker file
+	err := writeMarkerFileCommon(testMarkerFile, randomBytes(40), true)
+	require.NoError(t, err)
+
+	// Get length of file
+	fileInfo, err := os.Stat(testMarkerFile)
+	require.NoError(t, err)
+	originalSize := fileInfo.Size()
+
+	// Write a shorter marker file
+	err = writeMarkerFileCommon(testMarkerFile, randomBytes(25), true)
+	require.NoError(t, err)
+
+	// Get length of file
+	fileInfo, err = os.Stat(testMarkerFile)
+	require.NoError(t, err)
+	newSize := fileInfo.Size()
+
+	// Make sure shorter file has is smaller in length than
+	// the original long marker file
+	require.Less(t, newSize, originalSize)
+}
+
+func randomBytes(length int) []byte {
+	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZÅÄÖ" +
+		"abcdefghijklmnopqrstuvwxyzåäö" +
+		"0123456789" +
+		"~=+%^*/()[]{}/!@#$?|")
```

Comment:
Nit: Because it's just a test for the file size, I'd avoid special characters:
```suggestion
	chars := []rune("ABCDEFGHIJKLMNOPQRSTUVWXYZ" +
		"abcdefghijklmnopqrstuvwxyz" +
		"0123456789" +
		"~=+%^*/()[]{}/!@#$?|")
```